### PR TITLE
ATO-112- Add space point distortion classes

### PR DESCRIPTION
**Pull request by miranov25**
### Squashed version of Distortion branch (Rifki Sadikin)
JIRA discussion within ATO-112 and ATO-433
https://alice.its.cern.ch/jira/secure/attachment/40304/40304_spaceChargeUserManual.pdf

This pull request will replace previous pull request (https://github.com/alisw/AliRoot/pull/540)
* fixing non uniform grid distance function
* extending unit test
(pull request 540 will be closed)

### CODE REVIEW

**Comment by miranov25** [comment]
+test

**Comment by miranov25** [comment]
Hallo all.

Who can approve the commit? I see in first comment my name in the list https://github.com/alisw/AliRoot/pull/630#issuecomment-376822111
but the list is different in:
https://github.com/alisw/AliRoot/pull/630#issuecomment-378821460
For info I sent notification also for:
@rsadikin, @wiechula 

Marian

**Comment by chiarazampolli** [comment]
Hello Marian,

You have the rights to approve part of your commit, then for the rest there is the other people listed in the second message by github. I guess that you cannot approve the changes in TPC/TPCbase/CMakeLists.txt, as this is a more "core" file. @dberzano can confirm. 
I could in principle approve it, but I think it is better if @dberzano has a look.

Cheers,

Chiara

**Comment by miranov25** [comment]
Ciao Chiara.

Thanks for answer. 
The setup is misleading because people assume I can approve, while I can not.

While waiting for comments from Dario - could you approve test?
Unit test was not tested on all platforms

Marian

**Comment by chiarazampolli** [comment]
Ciao Marian,

I agree it is a bit misleading, as there is no explicit mention of who can approve what. You only see two lists, separated by a ";" in the first message from github. This is what I usually check... 

Cheers,

Chiara

**Comment by miranov25** [comment]
Dear all.

I understand that it take time to review changes, but could somebody do following:
* start test
* confirm that somebody plan to review/resp approve
We need these changes - in oredr to start space charge distortion calibration using currents for run 3

Regards
Marian

File: TPC/TPCbase/CMakeLists.txt [hunk 1]
```diff
@@ -109,6 +110,12 @@ set(SRCS
     AliTransform.cxx
     AliXRDPROOFtoolkit.cxx
     AliTPCSAMPAEmulator.cxx
+        AliTPC3DCylindricalInterpolator.cxx
```
Comment (dberzano) [inline TPC/TPCbase/CMakeLists.txt]: Indentation of new files in this CMakeLists.txt does not match the previous one, can you please fix?

**Comment by dberzano** [comment]
CMakeLists.txt cannot be modified freely and require admin approval as @chiarazampolli pointed out.

@miranov25 to me this is fine apart from the indentation glitch I have mentioned in the review. If you fix it I can approve, thanks!

**Comment by miranov25** [comment]
Hallo.
I changed indentation as requested. 
Local test on my ubuntu setup went fine:
````
104/105 Test #104: func_STAT_AliDrawStyleTest ...................   Passed    2.92 sec
        Start 105: func_TPCbase_AliTPCSpaceCharge3DDriftLine
105/105 Test #105: func_TPCbase_AliTPCSpaceCharge3DDriftLine ....   Passed   71.22 sec
100% tests passed, 0 tests failed out of 105
````
Marian

**Comment by dberzano** [comment]
+1 fine for me now, many thanks!

**Comment by miranov25** [comment]
Hallo Rifki.

Could you fix the problem reported in build/AliRoot/el6native. 
From the error message, I expect error will be eliminated  moving initalization of TF1s  from header file to constructor. 
Second type of error reported by compiler, I expect to be related to first problem

Thanks. 
Marian.

**Comment by miranov25** [comment]
Hallo.
Rifki made an fix, but we are not sure to test it on the  build/AliRoot/el6native
Could somebody enable +test?

**Comment by dberzano** [comment]
+test

**Comment by miranov25** [comment]
Test went fine. I think we should be able to merge.
Could somebody approve?

**Comment by miranov25** [comment]
Hallo all.
I checked the the log file for the test wich failed:
https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/630/3bf5492b174926a1b37e4b9c74f91eac24f17df6/build_AliRoot_release/fullLog.txt

The code failed in the MC simulation part:
````
Processing sim.C(1,"/mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/AliRoot/0-1/test/vmctest/gun/g3Config.C")...
````
Code failed in the MC-geant simulation part line (17470):
````
   NAME       LQSTOR NQSTRU  NQREF NQLINK LQMINR LQ2END JQDVLL JQDVSY NQFEND  LOW-1  LOW-N HIGH-1 HIGH-N SYST-1 SYST-N    END     
  /GCBANK/(FF685198)      1      2      2    164   5300      0      0     16      0     -1      0     -1      0     -1     -1     
 
 *** Break *** segmentation violation
`````
* I do not expect the problem is related to our change, but I would like to check.
Is it possible to get stacktrace, resp. coredump for the process which failed.

**Comment by miranov25** [comment]
One more question - were there some other recent test which failed in the simuation part?

**Comment by shahor02** [comment]
Hello @miranov25 
I've seen this Zebra crash in a few PRs, usually it disappears, as it seems to be a case here too.
Cheers,
 Ruben

**Comment by shahor02** [comment]
+1

**Pull request merged**